docs/workflow/trimming/feature-switches.md

```diff
@@ -14,6 +14,7 @@ configurations but their defaults might vary as any SDK can set the defaults dif
 | VerifyDependencyInjectionOpenGenericServiceTrimmability | Microsoft.Extensions.DependencyInjection.VerifyOpenGenericServiceTrimmability | When set to true, DependencyInjection will verify trimming annotations applied to open generic services are correct. |
 | _AggressiveAttributeTrimming | System.AggressiveAttributeTrimming | When set to true, aggressively trims attributes to allow for the most size savings possible, even if it could result in runtime behavior changes |
 | _ComObjectDescriptorSupport | System.ComponentModel.TypeDescriptor.IsComObjectDescriptorSupported | When set to true, supports creating a TypeDescriptor based view of COM objects. |
+| _DataSetXmlSerializationSupport | System.Data.DataSet.XmlSerializationSupport | When set to false, DataSet implementation of IXmlSerializable will throw instead of using trim-incompatible XML serialization. |
```

Comment:
Do we have some process to decide if something is a public or non-public switch? The "has been disabled in the app configuration" in the exception message sounds like something that we might want to leave configurable. If someone turns it on, my understanding is they'll get trimming warnings, so it should be fine.

Reply:
My question is more whether someone could have been using it with TrimMode=partial and wants to re-enable it. (My assumption is that this doesn't reflect on our IsTrimmable assemblies, but on user code.)

We do document BuiltInComInteropSupport and enabling that one is a lot more scarier. (I always wondered why we did that.)

Reply:
I'm questioning this because this might be the first feature switch PR that I'm tagged on adding a non-public switch. We have others that feel in the same category and are not underscored (`JsonSerializerIsReflectionEnabledByDefault` comes to mind). Enabling them with full trimming will likely break the app but the story might be different with partial trimming.

It comes down to whether we would consider this supported with partial trimming (to the extent that we say it's supported for JsonSerializerIsReflectionEnabledByDefault, EnableCppCLIHostActivation, EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization, etc.).

https://github.com/dotnet/docs/issues/41739#issuecomment-2225805786 touched on that a bit but I think we'll want to have some better guidelines around this.

I don't feel strongly however. I'm not a fan of warning-disabled partial trimming, but we have appmodels that still set it.